apps/whale-api/src/module.api/poolpair.fees.controller.e2e.ts

```diff
@@ -0,0 +1,363 @@
+    await mintTokens(container, token)
+    await testing.generate(1)
+  }
+
+  await addPoolLiquidity(container, {
+    tokenA: 'DUSD',
+    amountA: 10,
+    tokenB: 'DFI',
+    amountB: 10,
+    shareAddress: await getNewAddress(container)
+  })
+
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'BTC',
+    amountA: 100,
+    tokenB: 'DFI',
+    amountB: 1000,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'ETH',
+    amountA: 10,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'DOGE',
+    amountA: 10,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  const oracleId = await container.call('appointoracle', [await testing.generateAddress(), [
+    { token: 'AAPL', currency: 'USD' }
+  ], 1])
+  await testing.generate(1)
+
+  await testing.rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '1.5@AAPL', currency: 'USD' }] })
+  await testing.generate(1)
+
+  await container.call('setloantoken', [{
+    symbol: 'AAPL',
+    name: 'APPLE',
+    fixedIntervalPriceId: 'AAPL/USD',
+    mintable: true,
+    interest: new BigNumber(0.01)
+  }])
+  await testing.generate(1)
+  await createPoolPair(container, 'AAPL', 'DUSD')
+  await mintTokens(container, 'AAPL')
+  await addPoolLiquidity(container, {
+    tokenA: 'AAPL',
+    amountA: 10,
+    tokenB: 'DUSD',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+
+  // dex fee set up
+  await container.call('setgov', [{
+    ATTRIBUTES: {
+      'v0/poolpairs/2/token_a_fee_pct': '0.005', // DUSD-DFI
+      'v0/token/9/dex_in_fee_pct': '0.001', // dAAPL
+      'v0/token/1/dex_out_fee_pct': '0.003', // DUSD
+
+      // no v0/poolpairs fee for dBTC
+      'v0/token/3/dex_in_fee_pct': '0.007', // dBTC
+      'v0/token/3/dex_out_fee_pct': '0.013', // dBTC
+
+      // no v0/poolpairs fee for dDOGE
+      'v0/token/7/dex_in_fee_pct': '0.011', // DOGE
+      'v0/token/7/dex_out_fee_pct': '0.009' // DOGE
+    }
+  }])
+  await container.generate(1)
+}
+
+describe('get best path - DEX burn fees', () => {
+  it('should use correct v0/token/{id}/dex_(in/out) from gov', async () => {
+    const paths1 = await controller.getBestPath('3', '7') // dBTC -> dDOGE
+    const paths2 = await controller.getBestPath('7', '3') // dDOGE -> dBTC
+    /*
+      dBTC(id:3) to DFI(id:0) = 0.007% dBTC dex fee
+      DFI(id:0) to dDOGE(id:7) = 0.009% dDOGE dex fee
+    */
+    expect(paths1.bestPath).toStrictEqual([
+      {
+        estimatedDexFees: [{
+          amount: '0.00700000',
+          token: {
+            displaySymbol: 'dBTC',
+            symbol: 'BTC'
+          }
+        }],
+        estimatedReturn: '10.00000000',
```

Comment:
is this denoted in dBTC or dDOGE?
I think it's possible to change the structure to something like

```typescript
bestPath: [
  {
    estimates: {
      dexFee: '0.00700000',
      return: '10.00000000'
      token: {
        displaySymbol: 'dBTC',
        symbol: 'BTC'
      }
    }
  }
]

Reply:
neither, it is DFI (tokenB)

 hmm the estimated dex fees should be an array because there could be a fee on both tokenA and tokenB and it's useful to add an estimatedReturn on `tokenA` 
[Sample e2e with 2 dex fees on one poolpair](https://github.com/JellyfishSDK/jellyfish/pull/1601/files/85dc86b23c325fa4c27c00d4845ae05702d82f02#diff-7f86947ae35f08cfa839fbbd670a46f364d1edda7e2f4398b6b23e54a52cf1cbR293)

Reply:
ah got it. so the `estimatedReturn` in the path is always denoted in tokenB (never tokenA)?

Reply:
yes, correct!